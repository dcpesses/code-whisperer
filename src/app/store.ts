import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import counterReducer from '@/features/redux-counter/counterSlice';
import modalCommandListReducer from '@/features/modal-command-list/modalSlice';
// import twitchReducer from '../features/twitch/twitchSlice';
import channelReducer from '@/features/twitch/channel-slice.js';
import userReducer from '@/features/player-queue/user-slice.js';
import queueReducer from '@/features/player-queue/queue-slice';
import todosReducer from '@/features/todos/todosSlice';

const reducer = {
  counter: counterReducer,
  channel: channelReducer,
  modal: modalCommandListReducer,
  queue: queueReducer,
  user: userReducer,
  todos: todosReducer,
};
export const store = configureStore({ reducer });

export function getStoreWithState(preloadedState?: RootState) {
  return configureStore({ reducer, preloadedState });
}

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
