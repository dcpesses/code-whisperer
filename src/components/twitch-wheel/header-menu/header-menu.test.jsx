/* eslint-env jest */
import {vi} from 'vitest';
import { Provider } from 'react-redux';
import { getStoreWithState } from '@/app/store';
import { fireEvent, render, screen } from '@testing-library/react';
import HeaderMenu from './index';

vi.mock('../../../../package.json', () => {
  return {
    ...vi.importActual('../../../../package.json'),
    version: '0.0.0'
  };
});
const storeState = {
  channel: {
    user: {
      id: '1',
      login: 'twitchstreamer',
      display_name: 'TwitchStreamer',
      type: '',
      broadcaster_type: '',
      description: 'description',
      profile_image_url: 'https://static-cdn.jtvnw.net/jtv_user_pictures/profile_image-300x300.png',
      offline_image_url: '',
      view_count: 0,
      created_at: '2019-11-18T00:47:34Z'
    },
    moderators: [{
    }],
    vips:[{
    }],
  },
  modal: {
    visible: false
  },
  user: {
    info: {
      id: '0',
      login: 'twitchuser',
      display_name: 'TwitchUser',
      type: '',
      broadcaster_type: '',
      description: 'description',
      profile_image_url: 'https://static-cdn.jtvnw.net/jtv_user_pictures/profile_image-300x300.png',
      offline_image_url: '',
      view_count: 0,
      created_at: '2019-11-18T00:47:34Z'
    },
    chatters: {
      twitchuser: {
        id: '0',
        login: 'twitchuser',
        display_name: 'TwitchUser',
        type: '',
        broadcaster_type: '',
        description: 'description',
        profile_image_url: 'https://static-cdn.jtvnw.net/jtv_user_pictures/profile_image-300x300.png',
        offline_image_url: '',
        view_count: 0,
        created_at: '2019-11-18T00:47:34Z'
      }
    },
    moderatedChannels: [{
      broadcaster_id: '1',
      broadcaster_login: 'TwitchStreamer',
      broadcaster_name: 'twitchstreamer',
    }],
    whisperStatus: {
      twitchuser: {
        login: 'twitchuser',
        response: {
          msg: 'Code sent to @TwitchUser',
          status: 204
        }
      },
    }
  }
};


describe('HeaderMenu', () => {
  let store;
  let props;
  beforeEach(()=>{
    store = getStoreWithState(storeState);
    props = {
      gamesList: {
        maxPlayersList: [4, 6, 7, 8, 9, 10, 12, 16, 20],
        validGames: {
          'Any Version': {
            Quiplash: {
              'Min players': 3,
              'Max players': 8,
              Variants: [
                'quiplash',
                'quip',
                'ql'
              ]
            }
          },
        }
      },
      parentState: {
        allowGameRequests: true,
        gameSelected: {
          name: '',
          longName: '',
          partyPack: '',
          'Min players': 1,
          'Max players': 16,
          Variants: [],
          username: '',
          time: 0,
          locked: false,
          chosen: false
        },
        messages: {},
        colors: [
          '#b0a4f9',
          '#bff0ff'
        ],
        counter: 0,
        history: [
          {
            name: '',
            longName: '',
            partyPack: '',
            'Min players': 1,
            'Max players': 16,
            Variants: [],
            username: '',
            time: 0,
            locked: false,
            chosen: false
          }
        ],
        logUserMessages: true,
        nextGameIdx: 0,
        settings: {
          enableRoomCode: true,
          enableSubRequests: false,
          customDelimiter: null
        },
        showOptionsMenu: false,
        showOptionsModal: false,
        showPlayerSelect: true,
        userLookup: {
          dcpesses: {
            'badge-info': null,
            badges: {
              broadcaster: '1',
              'twitch-recap-2023': '1'
            },
            'client-nonce': 'e4a0a659463e57c0af69041b9f1d66ea',
            color: '#1E90FF',
            'display-name': 'dcpesses',
            emotes: null,
            'first-msg': false,
            flags: null,
            id: 'ec7b4abc-ad1d-4f02-8318-749056db35b3',
            mod: false,
            'returning-chatter': false,
            'room-id': '473294395',
            subscriber: false,
            'tmi-sent-ts': '1711159950484',
            turbo: false,
            'user-id': '473294395',
            'user-type': null,
            'emotes-raw': null,
            'badge-info-raw': null,
            'badges-raw': 'broadcaster/1,twitch-recap-2023/1',
            username: 'dcpesses',
            'message-type': 'chat'
          }
        }
      },
      debugItems: [
        { label: 'Load Mock Game Requests' },
        { label: 'Load Mock Game & Player Requests' },
        { label: 'Log Debug Environment' },
        { label: 'Toggle User Message Logging' }
      ],
      items: [
        { label: 'View Chat Commands' }
      ],
      settings: {
        enableModeratedChannelsOption: false,
        enableRoomCode: true,
        enableSubRequests: false,
        customDelimiter: null
      },
      showOptionsMenu: false,
      twitchApi: {
        userInfo: {
          id: '0',
          login: 'twitchuser',
          display_name: 'TwitchUser',
          type: '',
          broadcaster_type: '',
          description: 'description',
          profile_image_url: 'https://static-cdn.jtvnw.net/jtv_user_pictures/profile_image-300x300.png',
          offline_image_url: '',
          view_count: 0,
          created_at: '2019-11-18T00:47:34Z'
        },
        debug: true
      }
    };
  });
  test('Should render without error', () => {
    const {container} = render(
      <Provider store={store}>
        <HeaderMenu {...props} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByLabelText('Toggle navigation'));
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText('Settings', {selector: 'a.settings-menu'}));
    expect(container).toMatchSnapshot();
    // TODO: determine proper queries
    // fireEvent.click(screen.getByRole('button', { name: /allows host to set a room code that can be whispered to players\./i}));
    // expect(container).toMatchSnapshot();
    // fireEvent.change(screen.getByRole('textbox', {name: 'Use Custom Delimiter:'}), {target: {value: ' | '}});
    // expect(container).toMatchSnapshot();
  });

  test('Should render with moderated channels menu', () => {
    props.settings.enableModeratedChannelsOption = true;
    const {container} = render(
      <Provider store={store}>
        <HeaderMenu {...props} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByLabelText('Toggle navigation'));
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText('Settings', {selector: 'a.settings-menu'}));
    expect(container).toMatchSnapshot();
  });
});

