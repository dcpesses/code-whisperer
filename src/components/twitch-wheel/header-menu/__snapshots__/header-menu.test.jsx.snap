// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderMenu > Should render without error 1`] = `
<div>
  <nav
    class="bg-body-tertiary mb-3 py-0 raleway-font navbar navbar-light"
    data-bs-theme="dark"
  >
    <div
      class="container-fluid"
    >
      <div
        class="dropdown"
        id="dropdown-moderated-channels"
        variant="link"
      >
        <button
          aria-expanded="false"
          class="text-decoration-none text-white dropdown-toggle btn btn-link btn-sm"
          id="dropdown-moderated-channels-toggle"
          type="button"
        >
          <span
            class="fw-semibold navbar-brand"
          >
            <img
              alt="dcpesses"
              class="rounded-circle navbar-pfp-img"
              src="profile_image-300x300.png"
            />
             
            dcpesses
          </span>
        </button>
      </div>
      <button
        aria-controls="navbar-options-menu"
        aria-label="Toggle navigation"
        class="border-0 rounded-0 navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        aria-labelledby="navbar-options-menu-label"
        class="raleway-font offcanvas offcanvas-end"
        id="navbar-options-menu"
      >
        <div
          class="offcanvas-header"
        >
          <div
            class="fw-bold fs-4 offcanvas-title h5"
            id="navbar-options-menu-label"
          >
            Options
          </div>
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="fw-medium offcanvas-body"
        >
          <div
            class="justify-content-end flex-grow-1 pe-3 fs-5 navbar-nav"
          >
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              Logout
            </a>
            <hr
              class="border-bottom my-2"
            />
            <a
              class="settings-menu nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              Settings
            </a>
            <div
              class="accordion-dark accordion accordion-flush collapse"
              id="settings-menu"
            >
              <div
                class="accordion-body"
              >
                <button
                  class="btn settings-menu btn btn-link"
                  id="enable-room-code"
                  title="Allows host to set a room code that can be whispered to players."
                  type="button"
                >
                  <input
                    readonly=""
                    role="switch"
                    type="checkbox"
                  />
                   
                  <span>
                    Enable Room Code
                  </span>
                </button>
                <button
                  class="btn settings-menu btn btn-link"
                  title="Replaces the !join command with a custom term to use to join the Interested queue."
                  type="button"
                >
                  <span
                    for="custom-join-command"
                  >
                    Use Custom Join Command: 
                  </span>
                  <input
                    class="form-control"
                    id="custom-join-command"
                    name="custom-join-command"
                    placeholder="e.g. !join"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                </button>
                <button
                  class="btn settings-menu btn btn-link"
                  title="Messages users in chat when they've successfully joined the Interested queue."
                  type="button"
                >
                  <input
                    readonly=""
                    role="switch"
                    type="checkbox"
                  />
                   
                  <span>
                    Show Join Confirmation Message
                  </span>
                </button>
                <button
                  class="btn settings-menu btn btn-link"
                  title="Replaces the !leave command with a custom term to use to leave all of the queues."
                  type="button"
                >
                  <span
                    for="custom-leave-command"
                  >
                    Use Custom Leave Command: 
                  </span>
                  <input
                    class="form-control"
                    id="custom-leave-command"
                    name="custom-leave-command"
                    placeholder="e.g. !leave"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                </button>
                <button
                  class="btn settings-menu btn btn-link"
                  title="Messages users in chat when they've successfully left the queues."
                  type="button"
                >
                  <input
                    readonly=""
                    role="switch"
                    type="checkbox"
                  />
                   
                  <span>
                    Show Leave Confirmation Message
                  </span>
                </button>
                <button
                  class="btn settings-menu btn btn-link"
                  title="Uses a custom character or emote to separate requests listed in the chat."
                  type="button"
                >
                  <span
                    for="custom-delimiter"
                  >
                    Use Custom Delimiter: 
                  </span>
                  <input
                    class="form-control"
                    id="custom-delimiter"
                    name="custom-delimiter"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                </button>
              </div>
            </div>
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              View Chat Commands
            </a>
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              Changelog
            </a>
            <div
              class="position-absolute bottom-0 start-0 end-0 pb-3 text-center"
              id="options-debug-menu-items"
            >
              <div
                class="dropup-center dropup"
                id="dropdown-debug-menu-items"
                variant="link"
              >
                <button
                  aria-expanded="false"
                  class="text-decoration-none dropdown-toggle btn btn-link btn-sm"
                  id="dropdown-debug-menu-items-toggle"
                  type="button"
                >
                  version 0.0.0
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`HeaderMenu > Should render without error 2`] = `
<div>
  <nav
    class="bg-body-tertiary mb-3 py-0 raleway-font navbar navbar-light"
    data-bs-theme="dark"
  >
    <div
      class="container-fluid"
    >
      <div
        class="dropdown"
        id="dropdown-moderated-channels"
        variant="link"
      >
        <button
          aria-expanded="false"
          class="text-decoration-none text-white dropdown-toggle btn btn-link btn-sm"
          id="dropdown-moderated-channels-toggle"
          type="button"
        >
          <span
            class="fw-semibold navbar-brand"
          >
            <img
              alt="dcpesses"
              class="rounded-circle navbar-pfp-img"
              src="profile_image-300x300.png"
            />
             
            dcpesses
          </span>
        </button>
      </div>
      <button
        aria-controls="navbar-options-menu"
        aria-label="Toggle navigation"
        class="border-0 rounded-0 navbar-toggler"
        data-margin-right=""
        style=""
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
    </div>
  </nav>
</div>
`;

exports[`HeaderMenu > Should render without error 3`] = `
<div>
  <nav
    class="bg-body-tertiary mb-3 py-0 raleway-font navbar navbar-light"
    data-bs-theme="dark"
  >
    <div
      class="container-fluid"
    >
      <div
        class="dropdown"
        id="dropdown-moderated-channels"
        variant="link"
      >
        <button
          aria-expanded="false"
          class="text-decoration-none text-white dropdown-toggle btn btn-link btn-sm"
          id="dropdown-moderated-channels-toggle"
          type="button"
        >
          <span
            class="fw-semibold navbar-brand"
          >
            <img
              alt="dcpesses"
              class="rounded-circle navbar-pfp-img"
              src="profile_image-300x300.png"
            />
             
            dcpesses
          </span>
        </button>
      </div>
      <button
        aria-controls="navbar-options-menu"
        aria-label="Toggle navigation"
        class="border-0 rounded-0 navbar-toggler"
        data-margin-right=""
        style=""
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
    </div>
  </nav>
</div>
`;
